### PR TITLE
docs(custom-resources): complete documentation for onEvent

### LINKED ISSUE #26892: aws-cdk-lib.custom_resources: unfinished comments
### Describe the issue

On this line https://github.com/aws/aws-cdk/blob/9cb395c48c0c04f6e38f4441a95f9a514fb4a08c/packages/aws-cdk-lib/custom-resources/lib/provider-framework/runtime/framework.ts#L22

It doesn't finish the sentence. `If an error occus...` then what? 

### Links

https://github.com/aws/aws-cdk/blob/9cb395c48c0c04f6e38f4441a95f9a514fb4a08c/packages/aws-cdk-lib/custom-resources/lib/provider-framework/runtime/framework.ts#L22

### PR DESCRIPTION
**Pull request by msambol**
Closes #26892.

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/aws-cdk-lib/custom-resources/lib/provider-framework/runtime/framework.ts [hunk 1]
```diff
@@ -19,7 +19,7 @@ export = {
  * Any lifecycle event changes to the custom resources will invoke this handler, which will, in turn,
  * interact with the user-defined `onEvent` and `isComplete` handlers.
  *
- * This function will always succeed. If an error occurs
+ * This function will always succeed. If an error occurs, it is logged.
```
Comment (msambol) [inline packages/aws-cdk-lib/custom-resources/lib/provider-framework/runtime/framework.ts:22]: I _think_ this is correct based on my understanding of the code.

File: packages/aws-cdk-lib/custom-resources/lib/provider-framework/runtime/framework.ts [hunk 2]
```diff
@@ -19,7 +19,7 @@ export = {
  * Any lifecycle event changes to the custom resources will invoke this handler, which will, in turn,
  * interact with the user-defined `onEvent` and `isComplete` handlers.
  *
- * This function will always succeed. If an error occurs
+ * This function will always succeed. If an error occurs, it is logged but an error is not thrown.
```
Comment (msambol) [inline packages/aws-cdk-lib/custom-resources/lib/provider-framework/runtime/framework.ts:22]: I _think_ this is correct based on my understanding of the code.

**Review by kaizencc — DISMISSED** [review]
Thanks! I also think this is correct :)

**Comment by kaizencc** [comment]
@msambol uhhh i was surprised that integ tests were failing due to this change but is your change bundled into assets being used by integ tests? If so, we have to update them. However, in this rare case, I am okay with you running `yarn integ --update-on-failed --dry-run` since you're not touching code. The only thing that should change in integ tests are the s3 asset hash

**Comment by msambol** [comment]
@kaizencc Updated for your recommendations. I had to do the following:

```
yarn integ-runner --directory packages/@aws-cdk --update-on-failed --> from root
yarn integ --update-on-failed --dry-run --> from @aws-cdk-testing/framework-integ/test
```

Let me know if that isn't right.

**Review by kaizencc — APPROVED** [review]

**Review by vinayak-kukreja — APPROVED** [review]

**Pull request merged**